content/english/java-tictactoe/answer-key.md

```diff
@@ -222,22 +223,56 @@ public class TicTacToe_Nuevo {
                 return "";
             }
         }
-        return "It's a TIE! Try it again";
+        return "It's a TIE! Try again";
     }
 }
 ```
 
 # activity-5: AI Computer Move
 ```java
-// reimplement getComputerMove()
-public static int getComputerMove(){
-    Random rand = new Random();
-    int pos = rand.nextInt(9) + 1;
-    while(playerMoves.contains(pos) || computerMoves.contains(pos)){
-        pos = rand.nextInt(9) + 1;
+public static int getComputerMoveAI(String[] curBoard){
+        int pos = 0;
+        int bestScore = Integer.MIN_VALUE;
+        for(int i = 1; i <= 9; i++){
+            if(curBoard[i - 1].equals(" ")){
+                curBoard[i - 1] = "O";
+                int score = minimax(curBoard, false);
+                curBoard[i - 1] = " ";
+                if(score > bestScore){
+                    bestScore = score;
+                    pos = i;
+                }
+            }
+        }
+        return pos;
+    }
+
+    public static int minimax(String[] curBoard, boolean isMaximizing){
+        String result = getWinner(curBoard);
+
+        // terminating state return score
+        if(result.contains("TIE")) return 0;
+        if(result.contains("won")) return -1;
+        if(result.contains("lost")) return 1;
+                
+        // recurrence case
+        int bestScore = (isMaximizing)? Integer.MIN_VALUE : Integer.MAX_VALUE;
+        String symbol = (isMaximizing)? "O" : "X";
+        for(int i = 1; i <=9; i++){
```

Comment:
Why not loop `int i = 0; i < 9; i++)` to avoid having to do `i-1` everywhere?

Reply:
Good point!